src/frontend/src/components/ChangeRequestDropdown.tsx

```diff
@@ -0,0 +1,82 @@
+import { MenuItem, Select, SelectChangeEvent } from '@mui/material';
+import { isWithinInterval, subDays } from 'date-fns';
+import { Control, Controller } from 'react-hook-form';
+import { AuthenticatedUser, ChangeRequest, wbsPipe } from 'shared';
+import { useAllChangeRequests } from '../hooks/change-requests.hooks';
+import LoadingIndicator from './LoadingIndicator';
+import { useCurrentUser } from '../hooks/users.hooks';
+
+interface ChangeRequestDropdownProps {
+  control: Control<any, any>;
+  name: string;
+}
+
+const ChangeRequestDropdown = ({ control, name }: ChangeRequestDropdownProps) => {
+  const user = useCurrentUser();
+  const { isLoading, data: changeRequests } = useAllChangeRequests();
+  if (isLoading || !changeRequests) return <LoadingIndicator />;
+
+  const filteredRequests = getChangeRequests(changeRequests, user);
+
+  const approvedChangeRequestOptions =
+    filteredRequests.map((cr) => ({
+      label: `${cr.crId} - ${wbsPipe(cr.wbsNum)} - ${cr.submitter.firstName} ${cr.submitter.lastName} - ${cr.type}`,
+      value: cr.crId
+    })) ?? [];
+
+  return (
+    <Controller
+      control={control}
+      name={name}
+      render={({ field: { onChange, value } }) => (
+        <Select
+          id="cr-autocomplete"
+          displayEmpty
+          renderValue={(value) => (value ? value : 'Change Request Id')}
+          value={value}
+          onChange={(event: SelectChangeEvent<number>) => {
+            const {
+              target: { value }
+            } = event;
+            if (value) onChange(value);
+          }}
+          size={'small'}
+          placeholder={'Change Request Id'}
+          sx={{ width: 200, textAlign: 'left' }}
+        >
+          {approvedChangeRequestOptions.map((option) => (
+            <MenuItem key={option.value} value={option.value}>
+              {option.label}
+            </MenuItem>
+          ))}
+        </Select>
+      )}
+    />
+  );
+};
+
+// Filter and sort change requests to display in the dropdown
+function getChangeRequests(changeRequests: ChangeRequest[], user: AuthenticatedUser): ChangeRequest[] {
+  const today = new Date();
+  const fiveDaysAgo = subDays(today, 5);
+
+  const filteredRequests = changeRequests.filter(
+    (cr) => cr.dateReviewed && cr.accepted && isWithinInterval(cr.dateReviewed, { start: fiveDaysAgo, end: today })
+  );
+
+  // The current user's CRs should be at the top
+  filteredRequests.sort((a, b) => {
+    const isSubmitterAUser = a.submitter.firstName === user.firstName && a.submitter.lastName === user.lastName;
+    const isSubmitterBUser = b.submitter.firstName === user.firstName && b.submitter.lastName === user.lastName;
+
+    if (isSubmitterAUser && isSubmitterBUser) return 0;
+    if (isSubmitterAUser) return -1;
+    if (isSubmitterBUser) return 1;
+
+    return a.crId - b.crId;
+  });
```

Comment:
Do you really need all this code to sort these properly? I think just returning -1 if a is the same as the current user should work the same

Reply:
Do you mean something like

```
if (isSubmitterAUser || isSubmitterBUser) return -1;
else return a.crId - b.crId;
```

Reply:
I also think in that case, if B's submitter is the same as the current user and A is not, returning -1 would cause sorting the crs in reverse order where the current user's cr, B comes after cr, A `[A, B]`.